linkerd2-proxy/src/main.rs

```diff
@@ -21,9 +21,9 @@ fn main() {
         }
     };
 
-    tokio_compat::runtime::current_thread::Runtime::new()
+    tokio::runtime::Runtime::new()
```

Comment:
It seems that this introduces a regression that i can catch via the benchmark suite: the proxy thread eventually deadlocks such that no new connections are accepted. the admin thread continues to operate normally.

Reply:
Hmm...that's pretty weird. I can look into it.

Reply:
It turns out that I must have accidentally been testing the threaded runtime. I don't see anything like this when i explicitly set the basic scheduler, so i'll add that here.

Reply:
Ahh, okay, that explains the deadlock...I am sure there are plenty of mutexen in e.g. metrics that only work with the assumption that they're not contended by multiple threads (or that rely on the write lock not being contended or something).